packages/next/src/client/components/request-async-storage.external.ts

```diff
@@ -20,10 +20,8 @@ export const requestAsyncStorage: RequestAsyncStorage =
 
 export function getExpectedRequestStore(callingExpression: string) {
   const store = requestAsyncStorage.getStore()
-  if (!store) {
-    throw new Error(
-      `Invariant: \`${callingExpression}\` expects to have requestAsyncStorage, none available.`
-    )
-  }
-  return store
+  if (store) return store
+  throw new Error(
+    `\`${callingExpression}()\` was called outside a request/render scope. Read more: https://nextjs.org/docs/messages/next-headers-wrong-context`
```

Comment:
no need to mention render as render is also under request scope, app router handing also not related to render

Reply:
As a user, this might not be obvious to me at first.

Technically yes, our App Router primitive is a "Route request", which has the two subcategories of `Route Handler request` and `page render".

I thought it might be useful information/easier to grasp, reading this in the error itself, since users might still think in the Pages Router separation model of API Routes (`api/`)/pages rendered by React.